src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -0,0 +1,184 @@
+
+    public ContainerExecDecorator(KubernetesClient client, String podName, String containerName, String path, AtomicBoolean alive, CountDownLatch started, CountDownLatch finished) {
+        this.client = client;
+        this.podName = podName;
+        this.containerName = containerName;
+        this.path = path;
+        this.alive = alive;
+        this.started = started;
+        this.finished = finished;
+    }
+
+    @Override
+    public Launcher decorate(final Launcher launcher, final Node node) {
+        return new Launcher.DecoratedLauncher(launcher) {
+            @Override
+            public Proc launch(ProcStarter starter) throws IOException {
+                launcher.getListener().getLogger().println("Executing shell script inside container [" + containerName + "] of pod [" + podName + "]");
+                watch = client.pods().withName(podName)
+                        .inContainer(containerName)
+                        .redirectingInput()
+                        .writingOutput(launcher.getListener().getLogger())
+                        .writingError(launcher.getListener().getLogger())
+                        .withTTY()
+                        .usingListener(new ExecListener() {
+                            @Override
+                            public void onOpen(Response response) {
+                                alive.set(true);
+                                started.countDown();
+                            }
+                            @Override
+                            public void onFailure(IOException e, Response response) {
+                                alive.set(false);
+                                e.printStackTrace(launcher.getListener().getLogger());
+                                started.countDown();
+                                finished.countDown();
+                            }
+
+                            @Override
+                            public void onClose(int i, String s) {
+                                alive.set(false);
+                                started.countDown();
+                                finished.countDown();
+                            }
+                        }).exec();
+
+                waitQuietly(started);
+
+                //We need to get into the project workspace.
+                //The workspace is not known in advance, so we have to execute a cd command.
+                watch.getInput().write(("cd " + path + NEWLINE).getBytes(StandardCharsets.UTF_8));
+                doExec(watch, launcher.getListener().getLogger(), getCommands(starter));
+                proc = new ContainerExecProc(watch, alive, finished);
+                return proc;
+            }
+
+            @Override
+            public void kill(Map<String, String> modelEnvVars) throws IOException, InterruptedException {
+                getListener().getLogger().println("Killing process.");
+                ContainerExecDecorator.this.close();
```

Comment:
You are ignoring the argument, so how can you support more than one command in a given container? I must be missing something.